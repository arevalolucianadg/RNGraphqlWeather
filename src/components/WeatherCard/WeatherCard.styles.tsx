import styled, {css} from 'styled-components/native';

import {global} from '../../styles/global-styles';

interface IsFavoriteProps {
  isFavorite: boolean;
}
export const WeatherCardWrapper = styled.TouchableOpacity<IsFavoriteProps>`
  background-color: ${props =>
    props.isFavorite ? props.theme.colors.primary : props.theme.colors.paper};
  border-radius: ${global.border.RadiusMd};
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  height: 100px;
  margin-bottom: 16px;
  padding: 16px;
`;

const textColor = css<IsFavoriteProps>`
  color: ${props =>
    props.isFavorite ? props.theme.colors.white : props.theme.colors.text};
  font-family: ${global.font.weightMedium};
`;

export const CityName = styled.Text<IsFavoriteProps>`
  ${textColor}
  font-family: ${global.font.weightMedium};
  font-size: 22px;
  margin-bottom: 8px;
`;
export const Info = styled.View`
  display: flex;
  flex-direction: row;
  align-items: flex-end;
`;
export const CityTemp = styled.Text<IsFavoriteProps>`
  ${textColor}
  font-family: ${global.font.weightBold};
  font-size: 18px;
  letter-spacing: 0.4px;
  margin-right: 8px;
`;
export const CitySummaryTitle = styled.Text<IsFavoriteProps>`
  ${textColor}
  font-family: ${global.font.weightLight};
  font-size: 14px;
`;

export const WeatherIcon = styled.Image`
  height: 70px;
  width: 70px;
`;
