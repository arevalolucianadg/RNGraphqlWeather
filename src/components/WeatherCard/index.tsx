import React, {useState} from 'react';
import {ListRenderItemInfo, View} from 'react-native';
import {useNavigation} from '@react-navigation/native';

import {
  CityName,
  Info,
  WeatherCardWrapper,
  CityTemp,
  CitySummaryTitle,
  WeatherIcon,
} from './WeatherCard.styles';

import ICON from '../../assets/png/weatherIcons';
import { WeatherInfo } from '../../graphql/interfaces';

interface WeatherCardProps {
  city: ListRenderItemInfo<WeatherInfo>;
}

const WeatherCard = ({city: {item}}: WeatherCardProps) => {
  
  const [isFavorite, setIsFavorite] = useState(false);
  
  const {id, name, weather: {summary, temperature}} = item;

  const navigation = useNavigation();

  return (
    <WeatherCardWrapper
      isFavorite={isFavorite}
      onPress={() => navigation.navigate('WeatherDetail', {cityId: id})}>
      <View>
        <CityName isFavorite={isFavorite}>{name}</CityName>
        <Info>
          <CityTemp isFavorite={isFavorite}>{temperature.actual.toFixed(1)}Â°</CityTemp>
          <CitySummaryTitle isFavorite={isFavorite}>
            {summary.title}
          </CitySummaryTitle>
        </Info>
      </View>
      <WeatherIcon source={ICON[`icon${summary.icon}`]} />
    </WeatherCardWrapper>
  );
};

export default WeatherCard;
