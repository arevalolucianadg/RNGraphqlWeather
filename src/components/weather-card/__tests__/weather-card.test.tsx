import React from 'react';
import { render } from '@testing-library/react-native';

import WeatherCard from '..';
import WithTheme from '../../../core/styles/theme-provider';

describe('weather card component', () => {
  const cityMock = {
    id: '2267057',
    name: 'Lisbon',
    weather: {
      summary: {
        icon: '03n',
        title: 'Clouds',
      },
      temperature: {
        actual: 15.81,
      },
    },
  };

  it('should render a snapshot of non-favorite component', () => {
    const result = render(
      <WithTheme>
        <WeatherCard isFavorite={false} city={cityMock} onPress={jest.fn()} />
      </WithTheme>,
    );

    expect(result.toJSON()).toMatchSnapshot();
  });

  it('should render a snapshot of favorite component', () => {
    const result = render(
      <WithTheme>
        <WeatherCard isFavorite city={cityMock} onPress={jest.fn()} />
      </WithTheme>,
    );

    expect(result.toJSON()).toMatchSnapshot();
  });

  it('should render the city name', () => {
    const result = render(
      <WithTheme>
        <WeatherCard isFavorite={false} city={cityMock} onPress={jest.fn()} />
      </WithTheme>,
    );

    expect(result.getByText(cityMock.name)).toBeDefined();
  });
});
