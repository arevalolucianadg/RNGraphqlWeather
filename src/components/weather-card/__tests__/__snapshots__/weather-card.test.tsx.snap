// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`weather card component should render a snapshot of favorite component 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#504ED3",
      "borderRadius": 8,
      "display": "flex",
      "flexDirection": "row",
      "height": 100,
      "justifyContent": "space-between",
      "marginBottom": 16,
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
    }
  }
>
  <View>
    <Text
      isFavorite={true}
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "WorkSans-Medium",
            "fontSize": 22,
            "marginBottom": 8,
          },
        ]
      }
    >
      Lisbon
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "display": "flex",
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        isFavorite={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "WorkSans-Bold",
              "fontSize": 18,
              "letterSpacing": 0.4,
              "marginRight": 8,
            },
          ]
        }
      >
        15.8
        °
      </Text>
      <Text
        isFavorite={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "WorkSans-Light",
              "fontSize": 14,
            },
          ]
        }
      >
        Clouds
      </Text>
    </View>
  </View>
  <Image
    source={
      Object {
        "testUri": "../../../assets/png/weatherIcons/night-cloudy.png",
      }
    }
    style={
      Array [
        Object {
          "height": 70,
          "width": 70,
        },
      ]
    }
  />
</View>
`;

exports[`weather card component should render a snapshot of non-favorite component 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1C1C24",
      "borderRadius": 8,
      "display": "flex",
      "flexDirection": "row",
      "height": 100,
      "justifyContent": "space-between",
      "marginBottom": 16,
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
    }
  }
>
  <View>
    <Text
      isFavorite={false}
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "WorkSans-Medium",
            "fontSize": 22,
            "marginBottom": 8,
          },
        ]
      }
    >
      Lisbon
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "display": "flex",
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        isFavorite={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "WorkSans-Bold",
              "fontSize": 18,
              "letterSpacing": 0.4,
              "marginRight": 8,
            },
          ]
        }
      >
        15.8
        °
      </Text>
      <Text
        isFavorite={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "WorkSans-Light",
              "fontSize": 14,
            },
          ]
        }
      >
        Clouds
      </Text>
    </View>
  </View>
  <Image
    source={
      Object {
        "testUri": "../../../assets/png/weatherIcons/night-cloudy.png",
      }
    }
    style={
      Array [
        Object {
          "height": 70,
          "width": 70,
        },
      ]
    }
  />
</View>
`;
